### PR TITLE
Utility for sycning with s3 and loading checkpoints from s3

### PR DESCRIPTION
**Pull request by mitchellnw**
This PR introduces two additional arguments, which are `--sync-s3` and `--sync-s3-frequency`. Recommended use is to do `--sync-s3 s3://<path-to-bucket>` and `--logs /scratch/logs` which is hopefully local ssd.

Then, as you run, you should see logs at `/scratch/logs/<name>` and `s3://<path-to-bucket>/<name>`. So you don't have to use the local file system.

This PR also supports loading from `s3://<path-to-checkpoint>` -- it's a bit slow but not too bad.

### CODE REVIEW

**Comment by rom1504** [comment]
What about using fsspec for the same thing to avoid being locked on S3 ?

**Comment by rom1504** [comment]
Can you add a comment it makes sense to use --sync-s3 with a local --logs in the readme ?

**Comment by mitchellnw** [comment]
Thanks for the comments, updated! The default is syncing via `aws s3 sync`.

**Comment by rom1504** [comment]
@mitchellnw could you resolve the merge conflict ?

do you think we should merge ?

**Comment by mitchellnw** [comment]
ah, yea I should make this work with the new auto-resume which is where the conflict is coming from (https://github.com/mlfoundations/open_clip/pull/303). then yes I think good to merge after that

**Comment by mitchellnw** [comment]
merge conflict fixed but need to add support for the resume = 'latest' feature

**Comment by mitchellnw** [comment]
Ok, should be good to go.

File: src/training/main.py [hunk 1]
```diff
@@ -49,9 +51,16 @@ def natural_key(string_):
     return [int(s) if s.isdigit() else s for s in re.split(r'(\d+)', string_.lower())]
 
 
-def get_latest_checkpoint(path: str):
+def get_latest_checkpoint(path: str, remote : bool):
     # as writen, this glob recurses, so can pick up checkpoints across multiple sub-folders
-    checkpoints = glob.glob(path + '**/*.pt', recursive=True)
+    if remote:
+        result = subprocess.run(["aws", "s3", "ls", path + "/"], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```
Comment (rom1504) [inline src/training/main.py:57]: would be much better to use fsspec to do that kind of listing

**Comment by rom1504** [comment]
looks ok, let's go

**Pull request merged**